tools/config-generator/main.go

```diff
@@ -130,6 +130,26 @@ type baseProwJobTemplateData struct {
 // ################ data definitions that are used for the prow config file generation ################
 // ####################################################################################################
 
+// outputter is a struct that directs program output and counts the number of write calls.
+type outputter struct {
+	sink  io.Writer
+	count int
+}
+
+// init initializes an outputter with a writer object and sets the write count to 0.
+func (o *outputter) init(writer io.Writer) {
```

Comment:
this would be better as a constructor à la
```
func newOutputter(writer io.Writer) outputter {
  return outputter{writer, 0}
```

Reply:
Done.